src/main/java/net/sacredlabyrinth/phaed/simpleclans/events/EconomyTransactionEvent.java

```diff
@@ -0,0 +1,78 @@
+package net.sacredlabyrinth.phaed.simpleclans.events;
+
+import org.bukkit.OfflinePlayer;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class EconomyTransactionEvent extends Event implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+
+    private final OfflinePlayer player;
+    private double amount;
+    private final Cause cause;
+    private final TransactionType transactionType;
+    private boolean cancelled;
+
+    public EconomyTransactionEvent(OfflinePlayer affected, double amount, @Nullable Cause cause, TransactionType transactionType) {
+        this.player = affected;
+        this.amount = amount;
+        this.cause = cause;
+        this.transactionType = transactionType;
+    }
+
+    public OfflinePlayer getPlayer() {
+        return player;
+    }
+
+    public double getAmount() {
+        return amount;
+    }
+    public void setAmount(double amount) {
+        this.amount = amount;
+    }
+
+    @Nullable
+    public Cause getCause() {
+        return cause;
+    }
+
+    public TransactionType getTransactionType() {
+        return transactionType;
+    }
+
+    @Override
+    public @NotNull HandlerList getHandlers() {
+        return handlers;
+    }
+
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return cancelled;
+    }
+
+    /**
+     * If cancelled, the transaction will result as denied.
+     */
+    @Override
+    public void setCancelled(boolean cancelled) {
+        this.cancelled = cancelled;
+    }
+
+    public enum Cause {
+        CLAN_CREATION, CLAN_VERIFICATION, ADD_MEMBER_TO_CLAN, CLAN_BANK, DISCORD_CREATION, PLAYER_KILLED,
```

Comment:
`CLAN_BANK` is excessive here, since `Clan#withdraw` does call `ClanBalanceUpdateEvent` ([example](https://github.com/RoinujNosde/SimpleClans/blob/02bedc558a15b48a2b58d11ddc0b75fc211f0714/src/main/java/net/sacredlabyrinth/phaed/simpleclans/commands/clan/BankCommand.java#L74)).
(I don't like that two events would be calling then and they returns and does particularly the same thing)

Same with [CLAN_REGROUP](https://github.com/RoinujNosde/SimpleClans/blob/02bedc558a15b48a2b58d11ddc0b75fc211f0714/src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/ClanManager.java#L1090).